bdd4j-junit5/src/main/java/org/bdd4j/BDD4jParameterResolver.java

```diff
@@ -1,36 +1,75 @@
 import org.junit.jupiter.api.extension.ExtensionContext;
 import org.junit.jupiter.api.extension.ParameterContext;
 import org.junit.jupiter.api.extension.ParameterResolutionException;
 import org.junit.jupiter.api.extension.ParameterResolver;
 
-import java.lang.reflect.InvocationTargetException;
-
 /**
  * A {@link ParameterResolver} that can be used in BDD4j tests to inject step instances.
  */
 public class BDD4jParameterResolver implements ParameterResolver
 {
+    private static Object getInstanceOfScenarioBuildersParameterizedType(Constructor<?> ctor, Type parameterizedType)
+        throws InstantiationException, IllegalAccessException, InvocationTargetException, ClassNotFoundException
+    {
+        var type = GenericTypeReflector.getExactParameterTypes(ctor, parameterizedType)[0];
+
+        return Thread.currentThread()
+                     .getContextClassLoader()
+                     .loadClass(type.getTypeName())
+                     .getConstructors()[0].newInstance();
+    }
+
+    private static Constructor<?> getScenarioBuilderConstructor()
+        throws NoSuchMethodException
+    {
+        return ScenarioBuilder.class
+            .getDeclaredConstructor(BDD4jSteps.class);
+    }
+
+    private static Object instantiateScenarioBuilder(Parameter scenarioBuilderParameter, Class<?> type)
+        throws NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException,
+        ClassNotFoundException
+    {
+        Constructor<?> ctor = getScenarioBuilderConstructor();
+        Object parameterizedTypeInstance = getInstanceOfScenarioBuildersParameterizedType(ctor,
+                                                                                          scenarioBuilderParameter.getParameterizedType());
+        //noinspection JavaReflectionInvocation
+        return type.getConstructor(BDD4jSteps.class)
+                   .newInstance(parameterizedTypeInstance);
+    }
+
     /**
      * {@inheritDoc}
      */
     @Override
     public Object resolveParameter(final ParameterContext parameterContext,
                                    final ExtensionContext extensionContext)
-            throws ParameterResolutionException
+        throws ParameterResolutionException
     {
         try
         {
-            return parameterContext.getParameter()
-                                   .getType()
-                                   .getConstructors()[0].newInstance();
-        } catch (final InstantiationException | IllegalAccessException | InvocationTargetException e)
+            var parameter = parameterContext.getParameter();
+            Class<?> type = parameter.getType();
+
+            if (type.equals(ScenarioBuilder.class))
+                return instantiateScenarioBuilder(parameter, type);
+
+            return type.getConstructors()[0].newInstance();
+
+        } catch (final InstantiationException | IllegalAccessException | InvocationTargetException |
```

Comment:
I feel like it could be easier on the eye to pull the different exception cases up to where they actually occur and just throw a `ParameterResolutionException` from there. But that's only the case, when the exception types are not repeated in different methods.